gusto/transport_equation.py

```diff
@@ -407,5 +408,5 @@ class EulerPoincare(VectorInvariant):
 
     def advection_term(self, q):
         L = super().advection_term(q)
-        L -= 0.5*div(self.test)*inner(q, self.ubar)*dx
+        L -= 0.5*div(self.test)*inner(self.P(q), self.ubar)*dx
```

Comment:
So you only need to project the test function for this one term, and not any of the others?

Reply:
Would it make sense for the *caller* of these term constructors to ensure the test function is projected?

Reply:
You are right, one could for example simply project the test and trial functions when ```eqn.advection_term``` is called in ```advection.py```, but I thought for now I should only implement the projection where I have tested it, i.e. in the Euler Poincare transport equation form.